src/utils/cosmosdb_model_versioned.ts

```diff
@@ -223,4 +207,35 @@ export abstract class CosmosdbModelVersioned<
         .map(_ => incVersion(_.version))
         .getOrElse(0 as NonNegativeInteger)
     );
+
+  /**
+   * Insert a document with a specific version
+   * @param o
+   * @param version
+   * @param requestOptions
+   */
+  private createNewVersion(
+    o: T,
+    version: NonNegativeInteger,
+    requestOptions?: RequestOptions
+  ): TaskEither<CosmosErrors, TR> {
+    const modelId = this.getModelId(o);
+    return super.create(
+      {
+        ...o,
+        id: generateVersionedModelId(modelId, version),
+        version
+      } as TN & RetrievedVersionedModel,
+      requestOptions
+    );
+  }
+
+  /**
+   * Strips off meta fields which are nor part of the base model definition
+   * @param o
+   */
+  private toBaseType(o: TR): T {
+    const { _etag, _rid, _self, _ts, id, version, ...n } = o;
+    return (n as unknown) as T;
```

Comment:
Can have Type `T` an `id` field?

Reply:
why?

Reply:
This method removes the `id` field from the TR. But maybe `id` is reserved in a versioned model.

Reply:
In versioned models `id` is a meta-data field, as it doesn't ship data and it's basically a Cosmos thing. I don't like the manual cast but I see no other way to skim unwanted fields.